subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -1023,9 +1266,11 @@ int nrf_cloud_device_status_encode(const struct nrf_cloud_device_status *const d
 	}
 
 	if (dev_status->modem) {
-		err = nrf_cloud_modem_info_json_encode(dev_status->modem, device_obj);
-		if (err) {
-			goto cleanup;
+		if (IS_ENABLED(CONFIG_MODEM_INFO)) {
```

Comment:
i don't think this check should be added (in case of `NRF_CLOUD_INFO_CLEAR`)

Reply:
removed the check